src/utils/CachingUtils.ts

```diff
@@ -1,7 +1,42 @@
+import { DEFAULT_CACHING_SAFE_LAG, DEFAULT_CACHING_TTL } from "../constants";
+import { CachingMechanismInterface, Deposit, Fill } from "../interfaces";
 import { assert } from "./LogUtils";
+import { composeRevivers, objectWithBigNumberReviver } from "./ReviverUtils";
+import { isDefined } from "./TypeGuards";
 
 export function shouldCache(eventTimestamp: number, latestTime: number, cachingMaxAge: number): boolean {
   assert(eventTimestamp.toString().length === 10, "eventTimestamp must be in seconds");
   assert(latestTime.toString().length === 10, "eventTimestamp must be in seconds");
   return latestTime - eventTimestamp >= cachingMaxAge;
 }
+
+/**
+ * Calls the cache's get method and returns the result if it is defined, otherwise returns undefined.
+ * @param key The key to get from the cache.
+ * @param cache The cache to get the key from.
+ * @returns The value associated with the key in the cache, or undefined if the key is not in the cache.
+ */
+export async function getDepositInCache(key: string, cache: CachingMechanismInterface): Promise<Deposit | undefined> {
+  const depositRaw = await cache.get<string>(key);
+  return isDefined(depositRaw) ? JSON.parse(depositRaw, composeRevivers(objectWithBigNumberReviver)) : undefined;
+}
+
+export async function setDepositInCache(
+  deposit: Deposit,
+  currentChainTime: number,
+  cache: CachingMechanismInterface,
+  expirySeconds = DEFAULT_CACHING_TTL
+): Promise<void> {
+  if (shouldCache(deposit.quoteTimestamp, currentChainTime, DEFAULT_CACHING_SAFE_LAG)) {
```

Comment:
It might be worth a comment to note that `deposit.quoteTimestamp` can be controlled by the user, within certain bounds. If `DEFAULT_CACHING_SAFE_LAG` was set to 1 hour, then a depositor could effectively force something into the cache immediately by setting their `quoteTimestamp` 1 hour in the past.

We do append the block timestamp to the `DepositWithBlock` object type, but that's not available in this case. On that, should we actually consider caching that additional information?

Reply:
This is outside the scope of this PR, but worth a check. Wdyt @nicholaspai would this be worth exploring in this cycle?

Reply:
Perfect - I also added an additional check just to make sure that the variables themselves are not outside of the range.

https://github.com/across-protocol/sdk-v2/pull/422/commits/e128a05c3179bc913b73cbf47aaa8d0ee73f2d2c